src/meta/src/hummock/manager/mod.rs

```diff
@@ -901,6 +942,13 @@ where
             )?;
         }
 
+        // A task heartbeat is removed IFF a task report has an associated context_id
+        // and its task_status has been committed.
+        if let Some(context_id) = context_id {
```

Comment:
After using cancel_compact_task in line 232, this Some(context_id) = context_id will never be true. I think cancel_compact_task can be called after line 232 succeeds.

Reply:
I think we should rename cancel_compact_task, it should be cancel_unassigned_compact_task. 

Btw, there is a function that cancels ALL unassigned tasks I believe.

Reply:
Anw the main thing is we need to simplify the interface, there is a lot of implicit assumptions about it's usage and the usage of report_compact_task atm

Reply:
> I think we should rename cancel_compact_task, it should be cancel_unassigned_compact_task.

In https://github.com/singularity-data/risingwave/pull/4932 I mean to make cancel_compact_task's semantic "cancel task anyway".

Reply:
I think we should require report_compact_task to provide the context_id, only cancel will use None for report_compact_task_impl

Reply:
`if let Some(assignee_context_id) = assignee_context_id {` ?

Reply:
Makes sense, we always use the actual state. I will make the cluster membership thread purge remaining heartbeats forcefully if 

I will add a test that when a context is removed, the task heartbeats are all removed.

Reply:
So the behaviour is to remove task heartbeat IFF reported and assignee is not None OR context is removed.